messaging/usecase/message_usecase.go

```diff
@@ -0,0 +1,64 @@
+package usecase
+
+import (
+	"chat/domain"
+	"context"
+	"time"
+)
+
+type messageUseCase struct {
+	timeout           time.Duration
+	messageRepository domain.MessageRepository
+	roomRepository    domain.RoomRepository
+}
+
+// NewMessageUseCase instantiates a
+func NewMessageUseCase(t time.Duration, mr domain.MessageRepository, rr domain.RoomRepository) domain.MessageUseCase {
+	return &messageUseCase{timeout: t, messageRepository: mr, roomRepository: rr}
+}
+
+func (u *messageUseCase) IsAuthorized(userID, roomID string) (authorized bool) {
+	ctx, cancel := context.WithTimeout(context.Background(), u.timeout)
+	defer cancel()
+
+	studentChatRooms, err := u.roomRepository.GetRoomsFor(ctx, userID)
+	if err != nil {
+		return false
+	}
+
+	// todo: this is temporary. Must be removed!
+	authorized = true
+	return
+	for _, room := range studentChatRooms.Rooms {
+		if roomID == room.RoomID {
+			authorized = true
+			break
+		}
+	}
+
+	return authorized
+}
+
+func (u *messageUseCase) SendMessage(ctx context.Context, message *domain.Message) error {
```

Comment:
rename method: saveMessage might be a more accurate description of what this method is doing